x-pack/plugins/security_solution/scripts/extract_tactics_techniques_mitre.js

```diff
@@ -67,29 +67,110 @@ const getSubtechniquesOptions = (subtechniques) =>
+  if (ref != null) {
+    return {
+      id: ref.external_id,
+      reference: ref.url,
+    };
+  } else {
+    return { id: '', reference: '' };
+  }
+};
+
+const extractTacticsData = (mitreData) => {
+  const tactics = mitreData
+    .filter((obj) => obj.type === 'x-mitre-tactic')
+    .reduce((acc, item) => {
+      const { id, reference } = getIdReference(item.external_references);
+
+      return [
+        ...acc,
+        {
+          displayName: item.name,
+          shortName: item.x_mitre_shortname,
+          id,
+          reference,
+        },
+      ];
+    }, []);
+
+  return sortBy(tactics, 'displayName');
+};
+
+const normalizeTacticsData = (tacticsData) => {
+  return tacticsData.map((data) => {
+    const { displayName, id, reference } = data;
+    return { name: displayName, id, reference };
+  });
+};
+
+const extractTechniques = (mitreData) => {
+  const techniques = mitreData
+    .filter(
+      (obj) =>
+        obj.type === 'attack-pattern' &&
+        (obj.x_mitre_is_subtechnique === false || obj.x_mitre_is_subtechnique === undefined)
+    )
+    .reduce((acc, item) => {
+      let tactics = [];
+      const { id, reference } = getIdReference(item.external_references);
+      if (item.kill_chain_phases != null && item.kill_chain_phases.length > 0) {
+        item.kill_chain_phases.forEach((tactic) => {
+          tactics = [...tactics, tactic.phase_name];
+        });
       }
-      return obj;
-    },
-    { id: '', reference: '' }
-  );
 
-const buildMockThreatData = (tactics, techniques, subtechniques) => {
+      return [
+        ...acc,
+        {
+          name: item.name,
+          id,
+          reference,
+          tactics,
+        },
+      ];
+    }, []);
+
+  return sortBy(techniques, 'name');
+};
+
+const extractSubtechniques = (mitreData) => {
+  const subtechniques = mitreData
+    .filter((obj) => obj.x_mitre_is_subtechnique === true)
+    .reduce((acc, item) => {
+      let tactics = [];
+      const { id, reference } = getIdReference(item.external_references);
+      if (item.kill_chain_phases != null && item.kill_chain_phases.length > 0) {
+        item.kill_chain_phases.forEach((tactic) => {
+          tactics = [...tactics, tactic.phase_name];
+        });
+      }
+      const techniqueId = id.split('.')[0];
+
+      return [
+        ...acc,
+        {
+          name: item.name,
+          id,
+          reference,
+          tactics,
+          techniqueId,
+        },
+      ];
+    }, []);
+
+  return sortBy(subtechniques, 'name');
+};
+
+const buildMockThreatData = (tacticsData, techniques, subtechniques) => {
   const subtechnique = subtechniques[0];
   const technique = techniques.find((technique) => technique.id === subtechnique.techniqueId);
-  const tactic = tactics.find(
-    (tactic) => tactic.name === startCase(camelCase(technique.tactics[0]))
-  );
+  const tactic = tacticsData.find((tactic) => tactic.shortName === technique.tactics[0]);
```

Comment:
Instead of finding the parent tactic by doing text transformations:

```js
  const tactic = tactics.find(
    (tactic) => tactic.name === startCase(camelCase(technique.tactics[0]))
  );
```

we can leverage the tactic's short name that's available in the source json metadata.

I think this is a more robust way to do that.